File: src/core/timetable/model/Timetable.ts
import UserLecture from './UserLecture';

export default interface Timetable {
    _id?: string;
    user_id: string;
    year: number;
    semester: number;
    title: string;
    lecture_list: UserLecture[];
    theme: number;
    is_primary: boolean;
    updated_at: number;
};

File: core/src/main/kotlin/com/wafflestudio/snu4t/timetables/data/TimeTable.kt
package com.wafflestudio.snu4t.timetables.data

import com.wafflestudio.snu4t.lectures.data.Lecture
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import org.springframework.data.mongodb.core.mapping.FieldType
import java.time.LocalDateTime

@Document("timetables")
data class TimeTable(
        @Id
        var id: String? = null,
        @Field("user_id", targetType = FieldType.OBJECT_ID)
        var userId: String,
        var year: Int,
        var semester: Int,
        @Field("lecture_list")
        var lectures: List<Lecture>,
        var title: String,
        var theme: Int,
        @Field("updated_at")
        var updatedAt: LocalDateTime,
)
